### PR TITLE
various pkgs: use https homepage when http redirects (github.io)

### PR DESCRIPTION
**Pull request by wdconinc**
This PR fixes some `homepage` attributes where the http link redirects to https anyway. Also fixes py-owslib homepage.

### CODE REVIEW

**Review by RMeli — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-h5io/package.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 class PyH5io(PythonPackage):
     """Python Objects Onto HDF5."""
 
-    homepage = "http://h5io.github.io"
+    homepage = "https://h5io.github.io"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-h5io/package.py]: I cannot confirm this URL. When I attempt to access it though a browser, I get `There isn't a GitHub Pages site here.`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Cannot confirm use of `https` protocol for `py-h5io`.

**Review by debog — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-h5io/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-h5io/package.py]: Let's go with plain github then.
```suggestion
    homepage = "https://github.com/h5io/h5io"
```

**Review by tldahlgren — APPROVED** [review]
(Re) confirmed the homepages.

**Pull request merged**